### PR TITLE
add json schema for graphql

### PR DESCRIPTION
**Pull request by pvormste**
This pr adds the json schema for graphql in its initial form.

It includes:
- configuration of graphql endpoint
- configuration of validation
- configuration of playground
- configuration of execution (proxyOnly as for now)

### CODE REVIEW

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -591,6 +527,75 @@ type RequestSigningMeta struct {
 	SignatureHeader string   `bson:"signature_header" json:"signature_header"`
 }
 
+// GraphQLConfig is the root config object for GraphQL middlewares
+type GraphQLConfig struct {
+	// Enabled indicates if GraphQL should be enabled
+	Enabled bool `bson:"enabled" json:"enabled"`
+	// GraphQLApi is the API specific configuration
+	GraphQLApi GraphQLApi `bson:"api" json:"api"`
```
Comment (furkansenharputlu) [inline apidef/api_definitions.go]: Better to change to `GraphQLAPI` with all references

File: apidef/api_definitions.go [hunk 2]
```diff
@@ -591,6 +527,75 @@ type RequestSigningMeta struct {
 	SignatureHeader string   `bson:"signature_header" json:"signature_header"`
 }
 
+// GraphQLConfig is the root config object for GraphQL middlewares
```
Comment (furkansenharputlu) [inline apidef/api_definitions.go]: GoDoc convention is to finish with `.`

Better to finish all GoDoc with `.` in this PR.

File: apidef/api_definitions.go (hunk 1 above)
Comment (jensneuse) [reply apidef/api_definitions.go]: @furkansenharputlu what do you mean by "with all references"?

Comment (pvormste) [reply apidef/api_definitions.go]: In my understanding, it means to rename the type and field to `GraphQLAPI`, so it is consistent with the other definitions regarding naming.
I have changed it now.

File: apidef/api_definitions.go (hunk 2 above)
Comment (pvormste) [reply apidef/api_definitions.go]: Done.

**Pull request merged**